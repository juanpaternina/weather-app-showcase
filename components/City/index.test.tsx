import React from 'react';
import { render } from '@testing-library/react-native';
import { City } from './index';

describe('City Component', () => {
  test('renders correctly with given text', () => {
    const { getByText } = render(<City>Test City</City>);
    expect(getByText('Test City')).toBeTruthy();
  });

  test('has correct styles', () => {
    const { getByText } = render(<City>Styled City</City>);
    const cityElement = getByText('Styled City');
    expect(cityElement.props.style).toMatchObject({
      textAlign: 'center',
      width: 250,
      fontSize: 22,
      alignSelf: 'center',
    });
  });

  test('Snapshot', () => {
    const { toJSON } = render(<City>Test City</City>);
    expect(toJSON()).toMatchSnapshot();
  });
});
