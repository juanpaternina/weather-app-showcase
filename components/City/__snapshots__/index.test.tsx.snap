// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City Component Snapshot 1`] = `
<Text
  style={
    {
      "alignSelf": "center",
      "borderBottomLeftRadius": 16,
      "borderBottomRightRadius": 16,
      "borderTopLeftRadius": 16,
      "borderTopRightRadius": 16,
      "color": "black",
      "fontFamily": "bold",
      "fontSize": 22,
      "fontWeight": "bold",
      "textAlign": "center",
      "width": 250,
    }
  }
>
  Test City
</Text>
`;
